src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,130 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PLANNER_PLAN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'createdDateTime', 'owner'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.ownerGroupId) {
+      this.createPlan(args.options.ownerGroupId, args, logger, cb);
```

Comment:
Instead of repeating the call to `createPlan` twice, let's simplify the code, by ensuring that we have the groupId and then calling createPlan afterwards.